### DISCUSSION TITLE
Set max tasks for worker before exiting

### DISCUSSION THREAD

**Discussion by sharon-br** [celery/celery#7018] [Ideas]
# Checklist

- [x] I have checked the [issues list](https://github.com/celery/celery/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3A%22Issue+Type%3A+Feature+Request%22+)
  for similar or identical feature requests.
- [x] I have checked the [pull requests list](https://github.com/celery/celery/pulls?utf8=%E2%9C%93&q=is%3Apr+label%3A%22PR+Type%3A+Feature%22+)
  for existing proposed implementations of this feature.
- [x] I have checked the [commit log](https://github.com/celery/celery/commits/master)
  to find out if the if the same feature was already implemented in the
  master branch.
- [x] I have included all related issues and possible duplicate issues
  in this issue (If there are none, check this box anyway).

#### Related Issues

- None

#### Possible Duplicates

- None

# Brief Summary
The main idea is to allow settings some flag that will specify the number of tasks the worker should execute before exiting.
The reason to do so is to able to scale celery workers over Kubernetes in a reasonable matter.
You would create a pod with a worker set to consume 1 message and then exiting, in Kubernetes you will create a job for each message in the broker queue that will start that said pod.
The current way to do something like that is kinda ugly since the auto scaler of Kubernetes might decide to scale down a worker that is currently running and in long-running tasks that can be problematic

# Design

## Architectural Considerations

None

## Diagrams

N/A

## Alternatives

None

**Comment by quertenmont** [comment]
I have the same need.
